macos-selenium-mochajs/test.js

```diff
@@ -14,37 +14,115 @@
  * limitations under the License.
  */
 
-const { Builder, By, until } = require('selenium-webdriver');
-const { expect } = require('chai');
+const { Builder } = require('selenium-webdriver');
+const { expect } = require('expect');
 const chrome = require('selenium-webdriver/chrome');
+const fs = require('fs');
+const path = require('path');
+const {
+  install,
+  Browser,
+  resolveBuildId,
+  detectBrowserPlatform,
+  ChromeReleaseChannel,
+} = require('@puppeteer/browsers');
+const winston = require('winston');
+
+const logger = winston.createLogger({
+  level: process.env.LOG_LEVEL || 'info',
+  format: winston.format.cli(),
+  transports: [new winston.transports.Console()],
+});
+
+describe('Selenium chromedriver', function () {
+  // The deafult timeout of 2s is not enough for selenium tests.
+  this.timeout(60 * 1000);
 
-describe('ChromeDriver Network Conditions Regression', function() {
-  this.timeout(30000);
   let driver;
+  let chromedriverBuild;
+  let chromeBuild;
+
+  before(async function () {
+    // The chrome and chromedriver installation can take some time. Give 5
+    // minutes to install everything.
+    this.timeout(5 * 60 * 1000);
+    /**
+     * By default, the test uses the latest Chrome version. Replace with the
+     * specific Chromium version if needed, e.g. "144.0.7553.0" or use
+     * environment variable like `BROWSER_VERSION=142.0.7444.175`
+     */
+    const BROWSER_VERSION =
+      process.env['BROWSER_VERSION'] ??
```

Comment:
Hmm I was hoping we could keep the test script rather simple, so it's not confusing for an end user to know what to modify. Perhaps we should move these boilerplate to a separate file and just include, so that the test script itself is short (and sweet?)